icechunk/src/storage/mod.rs

```diff
@@ -106,4 +114,24 @@ pub trait Storage: fmt::Debug + private::Sealed {
         overwrite_refs: bool,
         bytes: Bytes,
     ) -> StorageResult<()>;
+
+    async fn list_chunks(
+        &self,
+    ) -> StorageResult<BoxStream<StorageResult<ListInfo<ChunkId>>>>;
```

Comment:
Can these impls be defaulted? The impls look the same mostly (except for the nested storage systems)

Reply:
great point, will try

Reply:
One tricky thing is that traits cannot be object safe if they have generic methods .... :facepalm:

Reply:
Ahhhhh right... not worth it then probably